shell/platform/linux/fl_engine.cc

```diff
@@ -20,7 +23,39 @@ G_DEFINE_QUARK(fl_engine_error_quark, fl_engine_error)
 
 G_DEFINE_TYPE(FlEngine, fl_engine, G_TYPE_OBJECT)
 
-// Callback from Flutter engine that are passed to the renderer
+// Contains the data required to be passed from the Flutter task thread to the
+// GLib thread
+typedef struct {
+  FlEngine* self;
+  FlutterTask task;
+} TaskData;
+
+TaskData* task_data_new(FlEngine* self, FlutterTask task) {
+  TaskData* data = static_cast<TaskData*>(g_malloc0(sizeof(TaskData)));
+  data->self = static_cast<FlEngine*>(g_object_ref(self));
+  data->task = task;
+
+  return data;
+}
+
+void task_data_free(gpointer d) {
+  TaskData* data = static_cast<TaskData*>(d);
+  g_object_unref(data->self);
+  free(data);
```

Comment:
Not familiar with these APIs, but shouldn't `g_malloc` be balanced by `g_free`?

Reply:
Good catch! It would run fine as is, but it should be a g_free for consistency..